src/words/tweet_tokenizer.jl

```diff
@@ -0,0 +1,271 @@
+             [ *\-.\)]*
+             \d{4}
+           )"""
+
+
+HTML_TAGS = r"""<[^>\s]+>"""
+ASCII_ARROWS = r"""[\-]+>|<[\-]+"""
+TWITTER_USERNAME = r"""(?:@[\w_]+)"""
+TWITTER_HASHTAGS = r"""(?:\#+[\w_]+[\w\'_\-]*[\w_]+)"""
+EMAIL_ADDRESSES = r"""[\w.+-]+@[\w-]+\.(?:[\w-]\.?)+[\w-]"""
+WORDS_WITH_APOSTROPHE_DASHES = r"""(?:[^\W\d_](?:[^\W\d_]|['\-_])+[^\W\d_])"""
+NUMBERS_FRACTIONS_DECIMALS = r"""(?:[+\-]?\d+[,/.:-]\d+[+\-]?)"""
+ELLIPSIS_DOTS = r"""(?:\.(?:\s*\.){1,})"""
+WORDS_WITHOUT_APOSTROPHE_DASHES = r"""(?:[\w_]+)"""
+
+
+
+# Core tokenizing regex
+WORD_REGEX = Regex("(?i:" * join([URLS.pattern
+                    PHONE_NUMBERS.pattern
+                    EMOTICONS_REGEX.pattern
+                    HTML_TAGS.pattern
+                    ASCII_ARROWS.pattern
+                    TWITTER_USERNAME.pattern
+                    TWITTER_HASHTAGS.pattern
+                    EMAIL_ADDRESSES.pattern
+                    WORDS_WITH_APOSTROPHE_DASHES.pattern
+                    NUMBERS_FRACTIONS_DECIMALS.pattern
+                    WORDS_WITHOUT_APOSTROPHE_DASHES.pattern
+                    ELLIPSIS_DOTS.pattern
+                    r"(?:\S)".pattern
+                    ], "|")
+                    * ")"
+                    )
+
+
+# WORD_REGEX performs poorly on these patterns:
+HANG_REGEX = r"""([^a-zA-Z0-9])\1{3,}"""
+
+# Regex for replacing HTML_Entities
+HTML_ENTITIES_REGEX = r"""&(#?(x?))([^&;\s]+);"""
+
+HANDLES_REGEX = r"""(?x)
+                (?<![A-Za-z0-9_!@#\$%&*])@(([A-Za-z0-9_]){20}(?!@))
+                |
+                (?<![A-Za-z0-9_!@#\$%&*])@(([A-Za-z0-9_]){1,19})(?![A-Za-z0-9_]*@)
+                """
+
+
+"""
+    replace_html_entities(input_text::AbstractString,
+                    remove_illegal=true) => (entities_replaced_text::AbstractString)
+
+Removes entities from text by converting them to their corresponding unicode character.
+`input_text::AbstractString` The string on which HTML entities need to be replaced
+`remove_illegal::Bool` If `True`, entities that can't be converted are
+removed. Otherwise, entities that can't be converted are kept "as
+is".
+Returns `entities_replaced_text::AbstractString`
+"""
+function replace_html_entities(input_text::AbstractString, remove_illegal=true)
+
+    function convert_entity(matched_text)
+
+        groups = match(HTML_ENTITIES_REGEX, matched_text).captures
+        entity_body = groups[3]
+        local number::Number = 0
+        if isempty(groups[1])
+            return(lookupname(HTML_Entities.default, entity_body))
+        else
+            if isempty(groups[2])
+                is_numeric = true
+                for i in entity_body
+                    if !isdigit(i)
+                        is_numeric = false
+                        break
+                    end
+                end
+                if is_numeric
+                    number = parse(Int, entity_body, base=10)
+                end
+            else
+                is_base_16 = true
```

Comment:
similar to the isnumeriic pattern.

```suggestion
                base_16_letters = ('a', 'b', 'c', 'd', 'e', 'f')
                is_base_16 = all(entity_body) do i
                         isdigit(i) || i in base_16_letters)
                end
```